_blogposts/2023-04-15-improving-interop.mdx

```diff
@@ -0,0 +1,383 @@
+---
+author: rescript-team
+date: "2023-04-15"
+title: Better interop with customizable variants
+badge: roadmap
+description: |
+  A tour of new capabilities coming in ReScript v11
+---
+
+ReScript v11 is around the corner, and it comes packed with new features that will improve interop with JavaScript/TypeScript. We will shine some light on each significant feature in a series of blog posts.
+
+Interop with JavaScript/TypeScript is a top priority for ReScript. Recently we've made some changes to the runtime representation of variants that'll allow you to use variants for a large number of new interop scenarios, zero cost. This is important, because variants is _the_ feature of ReScript, enabling great data modeling, pattern matching and more.
+
+_Pattern matching_ is really the key here. The changes we've made will enable you to write idiomatic ReScript leveraging pattern matching in many more scenarios.
+
+These changes will ship in `v11` of ReScript, and in this post we'll tell you all you need to know about them.
+
+## TLDR;
```

Comment:
TLDR will be unnecessary if you just get to the examples more quickly and have a simple bullet list in the intro with a few words about the highlights